packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2745,75 +2755,112 @@ class RenderMouseRegion extends RenderProxyBox {
   @visibleForTesting
   MouseTrackerAnnotation get hoverAnnotation => _hoverAnnotation;
 
-  void _updateAnnotations() {
+  MouseTrackerAnnotation _buildAnnotation() {
+    if ((_onEnter != null ||
+         _onHover != null ||
+         _onExit != null ||
+         opaque
+        ) &&
+        RendererBinding.instance.mouseTracker.mouseIsConnected) {
+      return MouseTrackerAnnotation(
+        onEnter: _handleEnter,
+        onHover: _handleHover,
+        onExit: _handleExit,
+        key: _annotationKey,
+      );
+    }
+    return null;
+  }
+
+  // About updating the properties of [RenderMouseRegion]:
+  //
+  // `RenderMouseRegion` has 2 categories of properties:
+  //
+  //  * "Loose properties": [RenderMouseRegion] cares if these properties are
+  //    effective (an effective property makes _annotationIsActive true), but
+  //    doesn't care about their exact values. Examples are all the callbacks;
+  //    their actual values are not important since they are proxied. If their
+  //    changes don't affect `_annotationIsActive`, `markNeedsPaint` can be
+  //    skipped.
+  //  * "Strict properties": [RenderMouseRegion] needs the exact values of these
+  //    properties. An example is `opaque`. Their changes must be followed by
+  //    `markNeedsPaint` in order to update their exact values to the tree.
```

Comment:
i'm not sure about the loose vs strict terminology... not sure i have a better suggestion though

Reply:
Strict -> exact?
Loose -> flexible?

Reply:
I've changed the structure so that these terms are no longer needed.